### ISSUE TITLE
 include mechanisms for table accessibility

### ISSUE THREAD

**Issue by shudson310** [docbook/docbook#83]
The HTML table model allows a better preparation of table structure for (especially screen reader) accessibility when transforming to HTML (and, by analogy, probably to PDF and other digitally accessed output formats too, though I haven't studied this yet).
The CALS table model does not come anywhere near the abilities of the HTML model in associating data cells with table headers.
Especially in complex table setups, a combined use of the HTML-model id and headers attributes is excellent for demystifying the structure of table data - where CALS does not provide any infrastructure that could be used to generate such structuring via XSLT.
Thus, including the HTML table model would cater to accessibility requirements in all publishing contexts transforming docBook content to be accessed online as HTML - and probably also other formats.
References:
Techniques for WCAG2.0: http://www.w3.org/TR/WCAG20-TECHS/H43.html
W3C research on complex tables: http://www.w3.org/html/wg/wiki/ComplexTables
WebAim Guide for accessible data tables: http://webaim.org/techniques/tables/data

In order to enhance accessibility for screen readers, the following changes
are proposed to the DocBook table model for DocBook v5.1 (and associated
schemas):
1. Allow @headers attribute on <entry>. Identifies one or more entry
elements that apply to its entry. The headers attribute contains an
unordered set of unique space-separated tokens, each of which is an ID
reference.
For example: 
<thead><row> 
<entry scope="col" id="col9a">12/12/2005</entry>
<entry scope="col" id="col9b">12/19/2005</entry>
<entry scope="col" id="col9c">12/26/2005</entry>
</row></thead> 
<row> 
<entry id="col8-1b">Actual</entry>
<entry headers="col9 col9a row1 col8-1b">11,123.45</entry>
<entry headers="col9 col9b row1 col8-1b">11,012.34</entry>
<entry headers="col9 col9c row1 col8-1b">10,987.64</entry>
</row>
2. Allow @scope attribute on <entry>. The scope attribute can contain one of
"row" or "col" or "rowgroup" or "colgroup", where:
* The row keyword, which maps to the row state. The row state means the
header cell applies to some of the subsequent cells in the same row(s).
* The col keyword, which maps to the column state. The column state means
the header cell applies to some of the subsequent cells in the same
column(s).
* The rowgroup keyword, which maps to the row group state. The row group
state means the header cell applies to all the remaining cells in the row
group. An entry element's scope attribute must not be in the row group state
if the element is not anchored in a row group.
* The colgroup keyword, which maps to the column group state. The column
group state means the header cell applies to all the remaining cells in the
column group. An entry element's scope attribute must not be in the column
group state if the element is not anchored in a column group.
* The auto state. The auto state makes the header cell apply to a set of
cells selected based on context.
The scope attribute's missing value default is the auto state.
<! ENTITY % tbl.entry.scope "scope (row|col|rowgroup|colgroup) #IMPLIED">
3. Add @rowheader to the colspec element, with possible values 'yes' or
'no'. There is no default value, but absence of the attribute implies 'no'.
4. Since there is no @summary in CALS, the table's summary should be allowed
in a caption. Ensure that <caption> is allowed within <table>.
This proposal is informed by the following resources (courtesy of Nathalie
Sequiera):
http://www.w3.org/html/wg/wiki/IssueTableHeaders
W3C HTML5 editor's draft (26 June 2012)
- scope: http://dev.w3.org/html5/spec/the-th-element.html#the-th-element
- headers: 
http://dev.w3.org/html5/spec/attributes-common-to-td-and-th-elements.html
Techniques for WCAG2.0: http://www.w3.org/TR/WCAG20-TECHS/H43.html
W3C research on complex tables: http://www.w3.org/html/wg/wiki/ComplexTables
WebAim Guide for accessible data tables:
http://webaim.org/techniques/tables/data

Accepted 01/2013

**Comment by kosek** [comment]
In a meanwhile CALS table model in DocBook has been extended with these a11y features. So merely recompiling Publishers schema against DocBook 5.2 schema should resolve this issue. Would that resolve original requirements?

**Comment by shudson310** [comment]
Yes, the new Publishers schema should be recompiled against the DocBook 5.2 schema to incorporate these mechanisms.
